trunk/modules/python/dataset/reader.py

```diff
@@ -5,7 +5,9 @@
 def normalize(v) :
     '''Normalize a vector in a naive manner.'''
     minimum, maximum = np.amin(v), np.amax(v)
-    return (v - minimum) / (maximum - minimum)
+    diff = maximum - minimum
+    denom = diff if diff != 0. else 1.
+    return (v - minimum) / denom
```

Comment:
This is needless calculation if the range is zero. Instead we should return v if diff == 0. (within epsilon) otherwise perform the normalization. This will save us opts, as I'm unsure if the compiler is smart enough to optimize this scenario